### DISCUSSION TITLE
Conditional setback/exclusion of specific participants

### DISCUSSION THREAD

**Discussion by ghost** [jspsych/jsPsych#1009] [Q&A] [answered]
Hi, 
I am trying to set up a chat study (I have a working chat-plugin) and am using the "html-button-response"-plugin as a consent window. Meaning that once two participants were matched through a specific algorithm and thus assigned to a chat-room, they are asked to click 'ok' in order to proceed to the chat-room. Now, I would like to have conditions for the case where a participant doesn't click 'ok'. More specifically I would like to exclude the unresponsive participant  from the study and to reassign the participant who was going to be matched with that unresponsive participant to a different participant. The latter would require to go back the timeline up until the trial, where the matching algorithm assigns partners. This leads me to the following two questions:
1. How can I kick one specific particular participant out of the study? I know that an experiment can be ended through `jsPsych.endExperiment(...)` - function, but that would end the experiment for everyone, right?
2. How can I set one specific participant back to a certain point in the timeline? I thought about a combination of nesting a timeline and looping it depending on a specific condition, but I couldn't make it work so far.

I hope that I was clear enough and will appreciate any help or hint.
Thanks a lot!

**Comment by jodeleeuw** [comment, marked as answer]
Hi @GGM652,

How are you managing the synchronous aspects of the experiment, i.e., relaying messages from one participant to another?

**Reply by ghost** [reply]
Hi @jodeleeuw  and thanks a lot for getting back to me. I'm sorry but it seems that I misunderstood my supervisor and that the question is thus obsolete. Apologies for any inconvenience this may have caused and thanks again.
